src/types.ts

```diff
@@ -70,14 +70,50 @@ export type Amount = {
     multiasset: Multiasset<Uint>,
 }
 
-export type DatumHash = FixlenBuffer<typeof DATUM_HASH_LENGTH>
+// Datum
+export enum DatumType {
+    HASH = 0,
+    INLINE = 1,
+}
+
+export type DatumHash = {
+    type: DatumType.HASH,
+    hash: FixlenBuffer<typeof DATUM_HASH_LENGTH>,
+}
+
+
+export type DatumInline = {
+    type: DatumType.INLINE,
+    bytes: Buffer,
+}
 
-export type TransactionOutput = {
+export type Datum = DatumHash | DatumInline
+
+// Transaction output
+export enum OutputType {
+    ARRAY_LEGACY,
+    MAP_BABBAGE,
+}
+
+export type LegacyTransactionOutput = {
```

Comment:
Let's not use "legacy" --- after some time, babbage might become legacy; "alonzo" is much more precise.

Reply:
Well, ledgerjs uses legacy, and that's what's written in CDDL, so we can keep it.